src/Kiota.Builder/Extensions/OpenApiUrlTreeNodeExtensions.cs

```diff
@@ -28,12 +28,14 @@ public static class OpenApiUrlTreeNodeExtensions {
                                     static (x, y) => $"{x}{GetDotIfBothNotNullOfEmpty(x, y)}{y}") :
                         string.Empty)
                     .ReplaceValueIdentifier();
-        public static string GetNodeNamespaceFromPath(this OpenApiUrlTreeNode currentNode, string prefix) =>
-            currentNode?.Path?.GetNamespaceFromPath(prefix);
+        public static string GetNodeNamespaceFromPath(this OpenApiUrlTreeNode currentNode, string prefix) {
+            ArgumentNullException.ThrowIfNull(currentNode);
+            return currentNode.Path.GetNamespaceFromPath(prefix);
+        }
         //{id}, name(idParam={id}), name(idParam='{id}'), name(idParam='{id}',idParam2='{id2}')
-        private static readonly Regex PathParametersRegex = new(@"(?:\w+)?=?'?\{(?<paramName>\w+)\}'?,?", RegexOptions.Compiled);
+        private static readonly Regex PathParametersRegex = new(@"(?:\w+)?=?'?\{(?<paramName>\w+)\}'?,?", RegexOptions.Compiled, TimeSpan.FromMilliseconds(100));
```

Comment:
Shouldn't the timeout be configurable? I imagine if we ever needed to change it, it would be easier if it was a constant that could be changed in one place